File: LeastNumberOfUniqueIntegers/findLeastNumOfUniqueInts.cpp
class Solution 
{
    public:
        int findLeastNumOfUniqueInts(vector<int>& arr, int k)
        {
            unordered_map<int, int> map;
            for (auto num : arr)
                map[num]++;
            vector<pair<int, int>> elems(map.begin(), map.end());
            sort(elems.begin(), elems.end(), [](const pair<int, int>& a, const pair<int, int>& b){
                return a.second < b.second;
            });
            int remaining = elems.size();
            for (auto& element : elems)
            {
                if (k >= element.second)
                {
                    k -= element.second;
                    remaining--;
                }
                else
                    break;
            }
            return remaining;
        }
};

File: LeastNumberOfUniqueIntegers/findLeastNumOfUniqueInts.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        count = Counter(arr)
        elems = sorted(count.items(), key=lambda x: x[1])
        remaining = len(elems)
        for key, freq in elems:
            if k >= freq:
                k -= freq
                remaining -= 1
            else:
                break
        return remaining
